### PR TITLE
Fix YamlScalar.span for values with trailing whitespace

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/yaml/issues/51

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @nex3

File: test/span_test.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:source_span/source_span.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+void _expectSpan(SourceSpan source, String expected) {
```
Comment (jakemac53) [inline test/span_test.dart]: nit: move to the bottom of the file please :)

File: test/span_test.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  ╵''',
+      );
+    });
+  });
+}
+
+void _expectSpan(SourceSpan source, String expected) {
+  final result = source.message('message');
+  printOnFailure("r'''\n$result'''");
+  expect(result, expected);
```
Comment (natebosch) [inline test/span_test.dart]: A failed `expect` will already print `result`... why do we need the `printOnFailure` here?

Comment (kevmoo) [reply test/span_test.dart]: I get an easy copy-paste output here. The "actually" print out is not helpful – at all.

I can remove it.

File: test/span_test.dart (hunk 1 above)
Comment (kevmoo) [reply test/span_test.dart]: done

**Comment by kevmoo** [comment]
> This generally lgtm although we should confirm that it won't cause a lot of test churn (probably just pull internally and try things out).

Verified internally. Zero issues.

**Pull request merged**